### PR TITLE
Missing ogl states

### PR DESCRIPTION
**Pull request by Tinob**
Add support for more missing functionality in glstate

### CODE REVIEW

File: src/video_core/engines/maxwell_3d.h [hunk 1]
```diff
@@ -553,7 +553,7 @@ class Maxwell3D final {
                     };
                     float depth_range_near;
                     float depth_range_far;
-                } viewport[NumViewports];
+                } viewports[NumViewports];
```
Comment (lioncash) [inline src/video_core/engines/maxwell_3d.h]: std::array

File: src/video_core/engines/maxwell_3d.h [hunk 2]
```diff
@@ -581,9 +581,10 @@ class Maxwell3D final {
                         BitField<0, 16, u32> min_y;
                         BitField<16, 16, u32> max_y;
                     };
-                } scissor_test;
+                    u32 fill;
+                } scissor_test[NumViewports];
```
Comment (lioncash) [inline src/video_core/engines/maxwell_3d.h]: ditto

File: src/video_core/engines/maxwell_3d.h (hunk 1 above)
Comment (Tinob) [reply src/video_core/engines/maxwell_3d.h]: that was tyhe original code, but caused compilation error on gcc so changed it in 13e5e15 to []

Comment (wwylele) [reply src/video_core/engines/maxwell_3d.h]: @Tinob You need to move the definition of `ViewportTransform` and `ViewPort` out of the anonymous union/struct to make it work.

Comment (Tinob) [reply src/video_core/engines/maxwell_3d.h]: done

**Pull request merged**